libcantera-devel fails on Windows

While `conda install -c cantera/label/dev libcantera-devel` works nicely on a Linux system, it fails on Windows.

Steps taken on my system:

```
(base) conda create -n cantera-dev scons numpy cython ruamel_yaml libboost git ipython
```
and on `conda-dev`
```
conda install -c cantera/label/dev cantera
conda install -c cantera/label/dev libcantera-devel
```

I tried this both with and without Visual Studio (2019) environments set up. In either case, the critical error appears to be this:

```
C:\Users\ischoegl>pushd "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\"
The system cannot find the path specified.
```
Which indicates that the conda package contains remnants of a Visual 2017 Enterprise build stack. I am attaching the full logging information: [libcantera-devel.log](https://github.com/Cantera/conda-recipes/files/3432758/libcantera-devel.log)

PS: the expected location for an anaconda installation should likely be dictated by the python system path variable as returned by `sysconfig.get_paths()['include']`

Some progress, but still not successful. The old error is gone, but now I'm getting stuck at
```
(cantera-dev) PS C:\Users\ischo\GitHub\ctapp> python setup.py build_ext --inplace
running build_ext
building 'ctapp._ctapp' extension
creating build
creating build\temp.win-amd64-3.8
creating build\temp.win-amd64-3.8\Release
creating build\temp.win-amd64-3.8\Release\ctapp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ictapp -IC:\Users\ischo\anaconda3\envs\cantera-dev\lib\site-packages\numpy\core\include -IC:\Users\ischo\anaconda3\envs\cantera-dev -IC:\Users\ischo\anaconda3\envs\cantera-dev\include -IC:\Users\ischo\anaconda3\envs\cantera-dev\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /EHsc /Tpctapp/_ctapp.cpp /Fobuild\temp.win-amd64-3.8\Release\ctapp/_ctapp.obj
_ctapp.cpp
C:\Users\ischo\anaconda3\envs\cantera-dev\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
ctapp/_ctapp.cpp(672): fatal error C1083: Cannot open include file: 'cantera/cython/funcWrapper.h': No such file or directory
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.26.28801\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
```
Per our previous conversation, the headers should be in `${CONDA_ENV_DIR}`. Alas, searching for the offending file ('funcwrapper.h`) results in nothing.

Error still persists on 2.5.0b1 (setting up a conda environment `cantera-test` similar to [here](https://gist.github.com/ischoegl/ab777f871a48d7f40ecc32c1c849faa3), but just using conda to install cantera, rather than compiling from scratch). FWIW, here's some of the error output when I try to compile [ctapp](https://github.com/ischoegl/ctapp):
```
[...]
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ictapp -IC:\Users\ischo\miniconda3\envs\cantera-test\lib\site-packages\numpy\core\include -IC:\Users\ischo\miniconda3\envs\cantera-test -IC:\Users\ischo\miniconda3\envs\cantera-test\include -IC:\Users\ischo\miniconda3\envs\cantera-test\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /EHsc /Tpctapp/_ctapp.cpp /Fobuild\temp.win-amd64-3.8\Release\ctapp/_ctapp.obj
    _ctapp.cpp
    C:\Users\ischo\miniconda3\envs\cantera-test\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
    ctapp/_ctapp.cpp(679): fatal error C1083: Cannot open include file: 'cantera/cython/funcWrapper.h': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\ischo\miniconda3\envs\cantera-test\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ischo\\GitHub\\ctapp\\setup.py'"'"'; __file__='"'"'C:\\Users\\ischo\\GitHub\\ctapp\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.
```
As before, searching for `funcWrapper.h` across my entire `miniconda3` main folder (which should be where things get installed) yields nothing.

No surprises, the issue persists.
```
C:\Users\ischo\miniconda3\envs\cantera-test\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
ctapp/_ctapp.cpp(679): fatal error C1083: Cannot open include file: 'cantera/cython/funcWrapper.h': No such file or directory
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.28.29333\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
```

fwiw, I just tested an installation of headers and libraries from Cantera compiled in a `conda` environment on Windows. While the current `main` (status right before `2.6.0b1`) results in correct installations, the libraries are potentially still not usable by third-party software, mostly due to symbols not being properly resolved. Here is one of the observed compilation errors
```
cantera.lib(AnyMap.obj) : error LNK2001: unresolved external symbol "public: __cdecl YAML::Emitter::Emitter(void)" (??0Emitter@YAML@@QEAA@XZ)
```
This points that successful compilation would also require resolution of Cantera/enhancements#140.

--- a/cantera-matlab/bld.bat
+++ b/cantera-matlab/bld.bat
@@ -30,10 +30,10 @@ IF ERRORLEVEL 1 EXIT 1
 :: prevent this package from clobbering any existing
 :: libcantera or Cantera Python interface files, possibly except the
 :: data files and the license file.
-ROBOCOPY "%STAGE_DIR%\%PREFIX_DIR%\samples\matlab" "%LIBRARY_PREFIX%\cantera\samples\matlab" /S /E
-ROBOCOPY "%STAGE_DIR%\%PREFIX_DIR%\matlab" "%LIBRARY_LIB%\cantera\matlab" /S /E
-ROBOCOPY "%STAGE_DIR%\%PREFIX_DIR%\data" "%LIBRARY_PREFIX%\cantera\data" /S /E
-ROBOCOPY "%STAGE_DIR%\%PREFIX_DIR%\doc" "%LIBRARY_PREFIX%\cantera\doc" /S /E
+ROBOCOPY "%STAGE_DIR%\%PREFIX_DIR%\share\cantera\samples\matlab" "%PREFIX%\share\cantera\samples\matlab" /S /E
+ROBOCOPY "%STAGE_DIR%\%PREFIX_DIR%\share\cantera\matlab" "%PREFIX%\share\cantera\matlab" /S /E
+ROBOCOPY "%STAGE_DIR%\%PREFIX_DIR%\share\cantera\data" "%PREFIX%\share\cantera\data" /S /E
+ROBOCOPY "%STAGE_DIR%\%PREFIX_DIR%\share\cantera\doc" "%PREFIX%\share\cantera\doc" /S /E
 
 echo ****************************
 echo MATLAB BUILD COMPLETED SUCCESSFULLY

--- a/cantera-matlab/build.sh
+++ b/cantera-matlab/build.sh
@@ -47,7 +47,7 @@ cp -R "${STAGE_DIR}/${PREFIX_DIR}/${CT_DATA_DIR}/" "${PREFIX}/${CT_DATA_DIR}"
 CT_DOC_DIR="share/cantera/doc"
 mkdir -p "${PREFIX}/${CT_DOC_DIR}"
 cp -R "${STAGE_DIR}/${PREFIX_DIR}/${CT_DOC_DIR}/" "${PREFIX}/${CT_DOC_DIR}"
-CT_LIB_DIR="lib/cantera/matlab"
+CT_LIB_DIR="share/cantera/matlab"
 mkdir -p "${PREFIX}/${CT_LIB_DIR}"
 cp -R "${STAGE_DIR}/${PREFIX_DIR}/${CT_LIB_DIR}/" "${PREFIX}/${CT_LIB_DIR}"
 

--- a/cantera-matlab/post-link.bat
+++ b/cantera-matlab/post-link.bat
@@ -15,5 +15,5 @@ IF EXIST "%MATLAB_STARTUP%.cantera.bak" FINDSTR /V "%% added by Cantera Conda In
 
 REM Add Cantera lines to the startup.m script
 ECHO. >> "%MATLAB_STARTUP%"
-ECHO setenv('CANTERA_DATA', '%PREFIX%\Library\cantera\data') %% added by Cantera Conda Installer for %CONDA_ENV% environment>> "%MATLAB_STARTUP%"
-ECHO addpath(genpath('%PREFIX%\Library\lib\cantera\matlab')) %% added by Cantera Conda Installer for %CONDA_ENV% environment>> "%MATLAB_STARTUP%"
+ECHO setenv('CANTERA_DATA', '%PREFIX%\share\cantera\data') %% added by Cantera Conda Installer for %CONDA_ENV% environment>> "%MATLAB_STARTUP%"
+ECHO addpath(genpath('%PREFIX%\share\cantera\matlab')) %% added by Cantera Conda Installer for %CONDA_ENV% environment>> "%MATLAB_STARTUP%"

--- a/cantera-matlab/post-link.sh
+++ b/cantera-matlab/post-link.sh
@@ -13,5 +13,5 @@ if [ -f "${MATLAB_STARTUP}" ]; then
 fi
 
 echo "
-addpath(genpath('${PREFIX}/lib/cantera/matlab')) % added by Cantera ${PKG_VERSION} Conda installer for ${CONDA_ENV} environment
+addpath(genpath('${PREFIX}/share/cantera/matlab')) % added by Cantera ${PKG_VERSION} Conda installer for ${CONDA_ENV} environment
 setenv('CANTERA_DATA', '${PREFIX}/share/cantera/data') % added by Cantera ${PKG_VERSION} Conda installer for ${CONDA_ENV} environment" >> ${MATLAB_STARTUP}

--- a/cantera/bld.bat
+++ b/cantera/bld.bat
@@ -12,7 +12,7 @@ SET /A CPU_USE=%CPU_COUNT% / 2
 IF %CPU_USE% EQU 0 SET CPU_USE=1
 
 SET "ESC_PREFIX=%PREFIX:\=/%"
-ECHO prefix="%ESC_PREFIX%/Library" >> cantera.conf
+ECHO prefix="%ESC_PREFIX%" >> cantera.conf
 ECHO boost_inc_dir="%ESC_PREFIX%/Library/include" >> cantera.conf
 
 CALL scons build -j%CPU_USE%

--- a/cantera/build_lib.bat
+++ b/cantera/build_lib.bat
@@ -3,8 +3,14 @@ echo "****************************"
 echo "DELETING files from devel except shared libraries and data"
 echo "****************************"
 
-rd /s /q %PREFIX%\Library\share\cantera\doc
-rd /s /q %PREFIX%\Library\share\cantera\samples
-rd /s /q %PREFIX%\Library\share\man
-rd /s /q %PREFIX%\Library\share
+rd /s /q %PREFIX%\share\cantera\doc
+rd /s /q %PREFIX%\share\cantera\samples
+rd /s /q %PREFIX%\share\man
 rd /s /q %PREFIX%\Library\include\cantera
+rd /s /q %PREFIX%\Scripts
+del /f %PREFIX%\Library\lib\cantera.lib
+del /f %PREFIX%\Library\lib\cantera_shared.lib
+
+echo "****************************"
+echo "DELETING files COMPLETED"
+echo "****************************"

--- a/cantera/build_lib.sh
+++ b/cantera/build_lib.sh
@@ -8,7 +8,7 @@ rm -rf $PREFIX/share/cantera/samples
 rm -rf $PREFIX/share/man
 rm -rf $PREFIX/include/cantera
 rm -rf $PREFIX/bin
-rm -rf $PREFIX/lib/pkg-config
+rm -rf $PREFIX/lib/pkgconfig
 rm -rf $PREFIX/lib/libcantera.a
 
 if [[ "$target_platform" == osx-* ]]; then

--- a/cantera/build_py.bat
+++ b/cantera/build_py.bat
@@ -2,20 +2,23 @@ echo ****************************
 echo PYTHON %PYTHON% BUILD STARTED
 echo ****************************
 
-COPY cantera.conf cantera.conf.bak
-DEL /F cantera.conf
-FINDSTR /V "python_package" cantera.conf.bak > cantera.conf
-DEL /F cantera.conf.bak
+:: Remove old Python build files, if they are present
+IF EXIST "build/python" RD /S /Q "build/python"
+IF EXIST "build/temp-py" RD /S /Q "build/temp-py"
+IF EXIST "%PREFIX%/bin/ck2yaml" DEL /F "%PREFIX%/bin/ck2yaml"
+IF EXIST "%PREFIX%/bin/cti2yaml" DEL /F "%PREFIX%/bin/cti2yaml"
+IF EXIST "%PREFIX%/bin/ctml2yaml" DEL /F "%PREFIX%/bin/ctml2yaml"
+IF EXIST "%PREFIX%/bin/yaml2ck" DEL /F "%PREFIX%/bin/yaml2ck"
 
-ECHO python_package='full' >> cantera.conf
 SET "ESC_PYTHON=%PYTHON:\=/%"
 ECHO python_cmd="%ESC_PYTHON%" >> cantera.conf
-CALL scons build
+
+CALL scons build python_package=y
+IF ERRORLEVEL 1 EXIT 1
+
+"%PYTHON%" -m pip install --no-deps --no-index --find-links=build\python\dist\ cantera
 IF ERRORLEVEL 1 EXIT 1
 
 echo ****************************
 echo PYTHON %PYTHON% BUILD COMPLETED SUCCESSFULLY
 echo ****************************
-
-"%PYTHON%" -m pip install --no-deps --no-index --find-links=build\python\dist\ cantera
-IF ERRORLEVEL 1 EXIT 1

--- a/cantera/build_py.bat
+++ b/cantera/build_py.bat
@@ -19,6 +19,8 @@ IF ERRORLEVEL 1 EXIT 1
 "%PYTHON%" -m pip install --no-deps --no-index --find-links=build\python\dist\ cantera
 IF ERRORLEVEL 1 EXIT 1
 
+ROBOCOPY "%SRC_DIR%\samples\python" "%PREFIX%\share\cantera\samples\python" /S /E
+
 echo ****************************
 echo PYTHON %PYTHON% BUILD COMPLETED SUCCESSFULLY
 echo ****************************

--- a/cantera/build_py.sh
+++ b/cantera/build_py.sh
@@ -16,6 +16,8 @@ ${BUILD_PREFIX}/bin/python `which scons` build python_package='y' python_cmd="${
 
 $PYTHON -m pip install --no-deps --no-index --find-links=build/python/dist cantera
 
+cp -r $SRC_DIR/samples/python $PREFIX/share/cantera/samples/python
+
 if [[ "$target_platform" == osx-* ]]; then
    VERSION=$(echo $PKG_VERSION | cut -da -f1 | cut -db -f1 | cut -dr -f1)
    file_to_fix=$(find $SP_DIR -name "_cantera*.so" | head -n 1)

